### PR TITLE
feat: Dialog when publishing/unpublishing event

### LINKED ISSUE #1223: Show Dialog after publishing and before unpublishing event
### Feature Request

**Feature Description**

1. After publishing event, user gets dialog to share event.
2. Before unpublishing event, user gets dialog with a warning that the event will be turned to draft.
   
**Would you like to work on the issue?**

Yes

### PR DESCRIPTION
**Pull request by Masquerade0097**
Fixes #1223

Checklist:

- [x] I have checked for PMD and check-style issues 
- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia) and my PR follows them.
- [x] My branch is up-to-date with the Upstream development branch.

Changes: 
1. Option to share event if published successfully.
2. Option to confirm to unpublish before unpublishing event.

Screenshots for the change:
ezgif-4-fc84b386e6

### CODE REVIEW

**Comment by srv-twry** [comment]
@Masquerade0097 We don't allow publishing an event which doesn't have any specified location. The server already has a check for it and will throw an error in this case. It will be great if you can check it in the app also. You can show the info that "You need to provide event location in order to publish it" or equivalent. You'll save a network request by doing so.

**Comment by srv-twry** [comment]
@Masquerade0097 Please link the issue as well.

File: app/src/main/java/org/fossasia/openevent/app/core/event/dashboard/EventDashboardPresenter.java [hunk 1]
```diff
@@ -90,11 +99,13 @@ public void toggleState() {
             .doFinally(() -> getView().showResult(event))
             .subscribe(updatedEvent -> {
                 event.state = updatedEvent.state;
-                final int successMessage = Event.STATE_PUBLISHED.equals(event.state) ? R.string.publish_success : R.string.draft_success;
-                getView().onSuccess(utilModel.getResourceString(successMessage));
+                if (Event.STATE_PUBLISHED.equals(event.state)) {
+                    getView().showShareDialog();
+                }else {
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/openevent/app/core/event/dashboard/EventDashboardPresenter.java]: space

**Comment by Masquerade0097** [comment]
@srv-twry @iamareebjamal I have added a dialog to ask the user to add location in edit event fragment before publishing (in case it is not set). Please review.

File: app/src/main/java/org/fossasia/openevent/app/common/Runnable.java [hunk 2]
```diff
@@ -0,0 +1,6 @@
+package org.fossasia.openevent.app.common;
+
+public interface Runnable {
+
+    void run();
+}
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/openevent/app/common/Runnable.java]: This is already defined in Java

File: app/src/test/java/org/fossasia/openevent/app/core/presenter/EventDashboardPresenterTest.java [hunk 3]
```diff
@@ -113,6 +118,8 @@ public void tearDown() {
 
     @Test
     public void shouldLoadEventAndAttendeesAndStatisticsAutomatically() {
+        when(databaseChangeListener.getNotifier()).thenReturn(PublishSubject.create());
```
Comment (iamareebjamal) [inline app/src/test/java/org/fossasia/openevent/app/core/presenter/EventDashboardPresenterTest.java]: Add this is setUp to avoid writing in each function

Comment (Masquerade0097) [reply app/src/test/java/org/fossasia/openevent/app/core/presenter/EventDashboardPresenterTest.java]: It is only required in 8/25 tests. That's why i added it explicitly in those methods

Comment (iamareebjamal) [reply app/src/test/java/org/fossasia/openevent/app/core/presenter/EventDashboardPresenterTest.java]: OK

**Comment by Masquerade0097** [comment]
@sridharjajoo Conflicts resolved.

**Pull request merged**